### ISSUE TITLE
p10k/zsh issue with prompt

### ISSUE THREAD

**Issue by michelle-slang** [warpdotdev/Warp#4427]
### Describe the bug

Since the last update (to v0.2024.02.27.08.01.stable_03), my p10k prompt connector line is broken. It used to look like this: 
<img width="266" alt="Screenshot 2024-03-01 at 6 59 16 AM" src="https://github.com/warpdotdev/Warp/assets/160188441/4145c12a-a95d-47f7-874a-6ff414945414">

It now looks like this:
<img width="403" alt="Screenshot 2024-03-01 at 7 17 32 AM" src="https://github.com/warpdotdev/Warp/assets/160188441/ee33ba58-4cd3-4ee2-b8d8-83bc41e9f318">

The prompt connector line is cut off and the text is all the way left-aligned instead of being aligned related to the connector line. 

### To reproduce

1. update Warp to the latest stable build (v0.2024.02.27.08.01.stable_03)
2. start/restart Warp
3. note where the text prompt is in relation to the p10k prompt connector line

### Expected behavior

The p10k prompt connector line should should not be cut off where it is and the text prompt should not be so far to the left. 

### Operating system

MacOS

### Operating system and version

Sonoma 14.2.1 (23C71)

### Shell Version

zsh 5.9 (x86_64-apple-darwin23.0)

### Current Warp version

v0.2024.02.27.08.01.stable_03

### Regression

Yes, this bug started recently or with an X Warp version

### Recent working Warp date

2/28/2024

### Does this block you from using Warp daily?

No

### Is this a Warp specific issue? (i.e. does it happen in Terminal, iTerm, Kitty, etc.)

Yes, this I confirmed this only happens in Warp, not other terminals.

### Warp Internal (ignore): linear-label:b9d78064-c89e-4973-b153-5178a31ee54e

None

**Comment by michelle-slang** [comment]
no, it happens every time I start or restart Warp, with any command. Also, this never happened before the last update. 

My config is attached.
[p10k.zsh.zip](https://github.com/warpdotdev/Warp/files/14502903/p10k.zsh.zip)

**Comment by dannyneira** [comment]
Thanks @michelle-slang, I was able to reproduce the issue with your p10k.zsh file. I'll let the team know, but as a possible workaround, backup your p10k.zsh (`cp ~/.p10k.zsh ~/.p10k.zsh.backup`), and then run `p10k configure` to configure the prompt from scratch, this may reset the p10k prompt to work correctly. If not then you can always revert back to your original prompt and wait for the fix with `mv ~/.p10k.zsh.backup ~/.p10k.zsh`. 

Also keep in mind, we're working on support for prompt in the same line as the input cursor, so you may not need to run `exec zsh` to get the prompt to work correctly. Hope this helps!

**Comment by michelle-slang** [comment]
That's great @dannyneira! I have reconfigured p10k several times hoping it might fix the issue but it doesn't change anything.

**Comment by Advait-M** [comment]
Hey @michelle-slang ! Quick update on this: we've identified the issue and the fix should be included in the next release (end of this week)! We'll post an update here once that's live.

Also wanted to note - we're working on better p10k support as a part of #2304 - I'd recommend following along there too! This will help support having the command on the same line as the last line of the prompt. Ideally, you'd want to use Warp without being in a "perma" subshell, since it limits Warp's features (when the subshell isn't Warpified) 😄.

**Comment by Advait-M** [comment]
Update: it should be fixed in the latest Warp release 😄 - could you please confirm @michelle-slang ?
